reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/MismatchedLaterality.java

```diff
@@ -3,199 +3,126 @@
+                                .add("notYetMembers").withDefaultValue(true)
+                                .add("alreadyMembers").withDefaultValue(true)
+                                .build()
+                )
+                .withTag(INT)
+                .build();
+    }
+
+    public void postInit() throws TermServerScriptException {
+        String[] tabNames = new String[]{
+                "Result",
+        };
+        String[] columnHeadings = new String[]{
+                "Identifier, FSN, SemTag, Member, Action, Comment"
+        };
+        super.postInit(tabNames, columnHeadings, false);
+
+        info("Populating FSN map for all concepts");
+        for (Concept c : gl.getAllConcepts()) {
+            if (c.isActive()) {
+                fsnMap.put(c.getFsn().toLowerCase(), c);
+            }
+        }
+    }
+
+    public void runJob() throws TermServerScriptException {
+        if (jobRun.getParamBoolean("notYetMembers")) {
+            // Process lateralisable Concepts not yet added to 723264001 |Lateralisable body structure reference set|.
+            String byLaterality = "( (<< 91723000 |Anatomical structure (body structure)| : 272741003 | Laterality (attribute) | = 182353008 |Side (qualifier value)|) MINUS ( * : 272741003 | Laterality (attribute) | = (7771000 |Left (qualifier value)| OR 24028007 |Right (qualifier value)| OR 51440002 |Right and left (qualifier value)|) ) )  MINUS (^ 723264001)";
+            String byHierarchy = "(( << 91723000 |Anatomical structure (body structure)| MINUS (* : 272741003 | Laterality (attribute) | = (7771000 |Left (qualifier value)| OR 24028007 |Right (qualifier value)| OR 51440002 |Right and left (qualifier value)|)))  AND (<  (^ 723264001)))   MINUS (^ 723264001)";
+            Set<Concept> notYetMembers = getConceptsByECL(byLaterality, byHierarchy);
+
+            reportOddNumberOfLateralisedChildren(notYetMembers, false);
+        }
+
+        if (jobRun.getParamBoolean("alreadyMembers")) {
+            // Process lateralisable Concepts already added to 723264001 |Lateralisable body structure reference set|.
+            String byMembership = "^ 723264001";
+            Set<Concept> alreadyMembers = getConceptsByECL(byMembership);
+
+            reportOddNumberOfLateralisedChildren(alreadyMembers, true);
+        }
+    }
+
+    private Set<Concept> getConceptsByECL(String... eclStatements) throws TermServerScriptException {
+        Set<Concept> concepts = new HashSet<>();
+        for (String eclStatement : eclStatements) {
+            if (jobRun.getParamBoolean("legacy")) {
+                concepts.addAll(findConcepts(eclStatement));
+            } else {
+                concepts.addAll(findConceptsWithoutEffectiveTime(eclStatement));
+            }
+        }
+
+        return concepts;
+    }
+
+    private void reportOddNumberOfLateralisedChildren(Set<Concept> lateralisableConcepts, boolean memberOfLateralisableReferenceSet) throws TermServerScriptException {
```

Comment:
If a Concept belongs to the lateralisable reference set, or should be, and there is an odd number of lateralised children, then write this to the report.